### DISCUSSION TITLE
How do I make a Windows build with CMake?

### DISCUSSION THREAD

**Discussion by TheMikirog** [efroemling/ballistica#262] [Q&A] [answered]
I want to mess with binary files and try out my edits, but the `cmake` command always makes a Linux build. For various reasons I cannot test it on Linux and instead I use WSL2 to test the game. 
Is there a way to test my build on Windows?

**Comment by efroemling** [comment, marked as answer]
> I want to mess with binary files and try out my edits, but the `cmake` command always makes a Linux build. For various reasons I cannot test it on Linux and instead I use WSL2 to test the game. 
> 
> Is there a way to test my build on Windows?

Technically it is possible to use CMake to generate windows builds (I think it generates an intermediate visual studio project, among other methods). However the limitation right now is that I need to generate the ballistica-internal libraries for windows, which are necessary to build the complete app. (the ballistica-internal library is the parts of the game that still aren't public for reasons such as security). Once I get a chance to automate windows builds of that library I'll just publish my own visual studio projects that people can use to build on windows. I'm hoping to prioritize that as soon as 1.6 is out the door. But sadly, for the immediate moment, building on mac or linux is probably the only viable option.

**Comment by efroemling** [comment]
Just swinging through to update this for posterity. It is now possible to assemble windows binaries (see https://github.com/efroemling/ballistica/discussions/284)
